lib/src/kbasesearchengine/events/StatusEvent.java

```diff
@@ -125,6 +128,14 @@ public Instant getTimestamp() {
         return isPublic;
     }
 
+    /** Get whether the corresponding record for this event should be overwritten
+     * if the record already exists in the index."
+     * @return whether to overwrite existing data.
+     */
+    public Optional<Boolean> isOverwriteExistingData() {
```

Comment:
Actually, is there any need for an optional here? can the builder assume that null = don't overwrite?

Reply:
Yes, but I would rather keep it consistent throughout. i.e. the field type, input type and output type should be same for these pojos. If we decide to go with the convention of just true/false with false as default, then there is no need to use the Optional wrapper in any of the pojos.

Reply:
I clarified with additional documentation. "An absent field indicates that the record must not be overwritten."

Reply:
Which, to reiterate, means that the semantics of false and absent are identical and therefore absent is redundant, but *shrug*